app/move/controllers/create/move-details.test.js

```diff
@@ -127,6 +140,145 @@ describe('Move controllers', function() {
           expect(nextSpy).to.be.calledOnceWithExactly()
         })
       })
+      context(
+        'appends different fields depending on location type',
+        function() {
+          let req
+          let res
+          let locationsStub
+          beforeEach(function() {
+            locationsStub = sinon
+              .stub(referenceDataService, 'getLocationsByType')
+              .withArgs('prison')
+              .resolves([
+                {
+                  id: '1',
+                  title: 'Albany',
+                },
+                {
+                  id: '2',
+                  title: 'Altcourse',
+                },
+              ])
+              .withArgs('court')
+              .resolves([])
+            req = {
+              t: sinon.stub().returns('__translated__'),
+              form: {
+                options: {
+                  fields: {
+                    to_location_court_appearance: {
+                      items: [],
+                    },
+                    date_type: {
+                      items: [{}, {}],
+                    },
+                    move_type: {
+                      items: [
+                        {
+                          value: 'court_appearance',
+                        },
+                        {
+                          value: 'prison_recall',
+                        },
+                      ],
+                    },
+                    additional_information: {},
+                    to_location_prison_recall: {},
+                  },
+                },
+              },
+            }
+            res = {
+              locals: {},
+            }
+          })
+          afterEach(function() {
+            locationsStub.resetHistory()
+          })
+
+          it('when the current location type is prison appends to_location_prison_recall as conditional', async function() {
```

Comment:
This is where `context` can come in useful for helping break down tests.

There are some examples in the rest of this file that may help. But something like this would be easier to read both in the code and the test outputs:

```javascript
context('when the current location type is prison', function() {
  beforeEach(async function() {
    req.session.currentLocation.location_type = 'prison'
    await controller.configure(req, res, nextSpy)
  })

  it('does thing one', function() {
    ...
  })

  it('does thing two', function() {
    ...
  })
})

context('when the current location type is police', function() {
  beforeEach(async function() {
    req.session.currentLocation.location_type = 'police'
    await controller.configure(req, res, nextSpy)
  })

  it('does thing one', function() {
    ...
  })

  it('does thing two', function() {
    ...
  })
})
```

Keeping `it` blocks to one or two `expect` statements also makes it much easier to debug tests when something fails.